lib/modules/ai_bot/personas/persona.rb

```diff
@@ -3,15 +3,28 @@
 module DiscourseAi
   module AiBot
     module Personas
-      def self.all
+      def self.all(user: nil)
         personas = [Personas::General, Personas::SqlHelper]
         personas << Personas::Artist if SiteSetting.ai_stability_api_key.present?
         personas << Personas::SettingsExplorer
         personas << Personas::Researcher if SiteSetting.ai_google_custom_search_api_key.present?
         personas << Personas::Creative
 
         personas_allowed = SiteSetting.ai_bot_enabled_personas.split("|")
-        personas.filter { |persona| personas_allowed.include?(persona.to_s.demodulize.underscore) }
+        personas =
+          personas.filter do |persona|
+            personas_allowed.include?(persona.to_s.demodulize.underscore)
+          end
+
+        if user
+          personas.concat(
+            AiPersona.all_personas.filter do |persona|
+              (persona.allowed_group_ids & user.group_ids).length > 0
```

Comment:
We have a shortcut method for this https://github.com/discourse/discourse/blob/e772bb9fce603d95bf23aca3d19576bbc22030f2/app/models/user.rb#L518-L520 

```ruby
user.in_any_groups?(persona.allowed_group_ids)
```